emba.sh

```diff
@@ -249,15 +248,56 @@ main()
     fi
   fi
 
+  if [[ $USE_DOCKER -eq 1 ]] ; then
+    if ! [[ $EUID -eq 0 ]] ; then
+      print_output "[!] Using emba with docker-compose requires root permissions" "no_log"
+      print_output "$(indent "Run emba with root permissions to use docker")" "no_log"
+      exit 1
+    fi
+    if ! command -v docker-compose > /dev/null ; then
+      print_output "[!] No docker-compose found" "no_log"
+      print_output "$(indent "Install docker-compose via apt-get install docker-compose to use emba with docker")" "no_log"
+      exit 1
+    fi
+
+    OPTIND=1
+    ARGS=""
+    while getopts a:A:cdDe:Ef:Fghik:l:m:sz OPT ; do
+      case $OPT in
+        D|f|i|l)
+          ;;
+        c)
+          print_output "" "no_log"
+          print_output "[-] Current docker version of emba does not support cwe-checker!" "no_log"
+          ;;
+        *)
+          export ARGS="$ARGS -$OPT"
+          ;;
+      esac
+    done
+
+    print_output "" "no_log"
+    print_output "[!] Emba initializes kali docker container.\\n" "no_log"
+
+    FIRMWARE="$FIRMWARE_PATH" LOG="$LOG_DIR" docker-compose run --rm emba -c "./emba.sh -l /log/ -f /firmware -i $ARGS"
+    D_RETURN=$?
+
+    if [[ $D_RETURN -eq 0 ]] ; then
+      if [[ $ONLY_DEP -eq 0 ]] ; then
```

Comment:
`[[ $D_RETURN -eq 0 && $ONLY_DEP -eq 0 ]]`

Reply:
Can be combined. But if only the dependencies are checked, the docker does not fail automatically.

Reply:
Then let it be :) I am still waiting for my setup to install and run the docker environment for testing. If it goes as well as the code looks, I'll merge it.